pkg/util/port/portlist/portlist_macos.go

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin && !ios
+// +build darwin,!ios
+
+package portlist
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"os/exec"
+	"sort"
+	"strings"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"go4.org/mem"
+)
+
+func init() {
+	newOSImpl = newMacOSImpl
+
+	// We have to run netstat, which is a bit expensive, so don't do it too often.
+	pollInterval = 5 * time.Second
+}
```

Comment:
Since we are importing the code and we can initialize the port implementation using our own way, I would like to suggest that rather than using `init()` functions we create a [`poller.init`](https://github.com/DataDog/datadog-agent/pull/25209/files#diff-ed48e04d7b4df89a302f3f8ba9834e238f835bbfa5a1b6c5475d28b2b889527cR78) for each runtime. That way we can avoid having to many `init()` functions and global variables.

I'm happy to pair if you want 😄